gazebo_ros/scripts/spawn_model.py

```diff
@@ -0,0 +1,312 @@
+        source = parser.add_mutually_exclusive_group(required=True)
+        source.add_argument(
+            '-file', type=str, metavar='FILE_NAME', help='Load model xml from file')
+        source.add_argument(
+            '-param', type=str, metavar='PARAM_NAME', help='Load model xml from ROS parameter')
+        source.add_argument('-database', type=str, metavar='MODEL_NAME',
+                            help='Load model XML from specified model in Gazebo Model Database')
+        source.add_argument('-stdin', action='store_true', help='Load model from stdin')
+        parser.add_argument(
+            '-model', required=True, type=str, metavar='MODEL_NAME', help='Name of model to spawn')
+        parser.add_argument('-reference_frame', type=str, default='',
+                            help='Name of the model/body where initial pose is defined.\
+                            If left empty or specified as "world", gazebo world frame is used')
+        parser.add_argument('-gazebo_namespace', type=str, default='/gazebo',
+                            help='ROS namespace of gazebo offered ROS interfaces. \
+                            Defaults to /gazebo/')
+        parser.add_argument('-robot_namespace', type=str, default=self.get_namespace(),
+                            help='change ROS namespace of gazebo-plugins')
+        parser.add_argument('-unpause', action='store_true',
+                            help='unpause physics after spawning model')
+        parser.add_argument('-wait', type=str, metavar='MODEL_NAME',
+                            help='Wait for model to exist')
+        parser.add_argument('-x', type=float, default=0,
+                            help='x component of initial position, meters')
+        parser.add_argument('-y', type=float, default=0,
+                            help='y component of initial position, meters')
+        parser.add_argument('-z', type=float, default=0,
+                            help='z component of initial position, meters')
+        parser.add_argument('-R', type=float, default=0,
+                            help='roll angle of initial orientation, radians')
+        parser.add_argument('-P', type=float, default=0,
+                            help='pitch angle of initial orientation, radians')
+        parser.add_argument('-Y', type=float, default=0,
+                            help='yaw angle of initial orientation, radians')
+
+        # TODO(shivesh): Wait for /set_model_configuration
+        # (https://github.com/ros-simulation/gazebo_ros_pkgs/issues/779)
+        # parser.add_argument('-J', dest='joints', default=[], action='append',
+        #     metavar=('JOINT_NAME', 'JOINT_POSITION'), type=str, nargs=2,
+        #     help='initialize the specified joint at the specified position')
+
+        parser.add_argument('-package_to_model', action='store_true', help='convert urdf \
+                            <mesh filename="package://..." to <mesh filename="model://..."')
```

Comment:
Did you give this parameter a try? How do you recommend I try it out?

Reply:
"Setting parameters from the command-line is currently only supported in the form of yaml files". This is what the [documentation](https://index.ros.org//doc/ros2/Tutorials/Node-arguments/) says. Should we do it?

Reply:
Oh sorry, I chose the word poorly. I didn't mean we should use parameters, I should have said "argument".

I just meant to ask how can we try this out to check it works. Did you create a dummy model which uses `<mesh filename="package://`?

Reply:
Oh. For that I printed just created a random urdf.
```
<?xml version="1.0"?>
<robot name="robot">
  <link name="base_link">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="10" />
      <inertia ixx="1" ixy="1" ixz="1" iyy="1" iyz="1" izz="1" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://robot_description/meshes/base_link.STL" />
      </geometry>
      <material name="">
        <color rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://robot_description/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
</robot>
```
And did `print(model_xml)` before line 171 and after 185 to see the change in the xml